src/cursor/cur_stat.c

```diff
@@ -429,6 +429,92 @@ __curstat_file_init(
     return (ret);
 }
 
+/*
+ * __page_stat_get --
+ *     Helper function used to initialize the statistics cursor for page stats.
+ */
+static int
+__page_stat_get(
+  WT_SESSION_IMPL *session, const char *uri, int64_t *byte_countp, int64_t *row_countp)
+{
+    WT_CKPT ckpt;
+    WT_DECL_RET;
+    WT_TABLE *table;
+    int64_t byte_count, max_row_count, row_count;
+    u_int i;
+
+    table = NULL;
+
+    WT_PREFIX_SKIP(uri, "checkpoint:");
+    if (WT_PREFIX_MATCH(uri, "file:")) {
+        if ((ret = __wt_meta_checkpoint(session, uri, NULL, &ckpt)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        *byte_countp = ckpt.ps.byte_count;
+        *row_countp = ckpt.ps.row_count;
+    } else if (WT_PREFIX_SKIP(uri, "table:")) {
+        if ((ret = __wt_schema_get_table(session, uri, strlen(uri), false, 0, &table)) != 0)
+            WT_ERR(ret == WT_NOTFOUND ? ENOENT : ret);
+
+        /*
+         * Set the number of rows as the maximum column group row count and sum the column group
+         * byte counts. Invalidate the table byte/row counts if any of the column group counts don't
+         * exist.
+         */
+        max_row_count = row_count = WT_STAT_NONE;
+        *byte_countp = 0;
+        for (i = 0; i < WT_COLGROUPS(table); i++) {
+            WT_ERR(__page_stat_get(session, table->cgroups[i]->source, &byte_count, &row_count));
+
+            if (byte_count == WT_STAT_NONE) {
+                *byte_countp = byte_count;
+                break;
+            } else
+                *byte_countp += byte_count;
+
+            if (row_count == WT_STAT_NONE) {
+                *row_countp = row_count;
+                break;
```

Comment:
After this break, later we replace the `*row_countp` with `max_row_count` and it is wrong.